addons/account/models/account_bank_statement.py

```diff
@@ -874,3 +874,4 @@ def _check_invoice_state(self, invoice):
 
     def button_confirm_bank(self):
         self.statement_id.button_confirm_bank()
+        return self.statement_id.id
```

Comment:
This will change the stable policy as it changes the value of what is returned (even if we previously did not return anything) and furthermore it does not fully fix the problem as I've tested the following case and it does not works as intended.

Use case: Create a statement with 2 lines, be sure that the ending balance is correctly entered then click on the reconcile button from the statement form view.
Reconcile both line, you are displayed with the rainbowman and a close statement button. Click the close statement button and it will :
1) not close the statement
2) open the bank statement form view without an id (so it is blank like when you click on the create button).

The reason for that is the following:
in the javascript we do
```js
closeStatement: function () {
        var self = this;
        return this._rpc({
                model: 'account.bank.statement.line',
                method: 'button_confirm_bank',
                args: [self.bank_statement_line_id.id],
            })
```
or self.bank_statement_line_id is defined in the reload function
`self.bank_statement_line_id = self.statement_line_ids.length === 1 ? {id: self.statement_line_ids[0], display_name: statement.statement_name} : false;`

and self.statement_line_ids comes directly from the context and contains the ids of both statement lines (so length === 2) (I've also notice by grepping the code that in some place we use statement lines ids in that context key and in some other place it seems we use a statement id...)

Anyway, this means that in this case we will call the `button_confirm_bank` without any args value as self.bank_statement_line_id is false, this means that we won't close the statement and we don't have a statement id to open the form view.

I believe it would probably be best to fix it like this:

in `def get_bank_statement_data(self, bank_statement_line_ids, srch_domain=[]):` which is in the reconciliation_widget.py file, change the following:

```diff
results.update({
            'statement_name': len(bank_statements_left) == 1 and bank_statements_left.name or False,
+            'statement_id': len(bank_statements_left) == 1 and bank_statements_left.id or False,
            'journal_id': bank_statements and bank_statements[0].journal_id.id or False,
            'notifications': []
        })
```
This allows us to have the value of a statement_id if we only have one statement id.
So in the javascript we can now do smth like:
```diff
closeStatement: function () {
        var self = this;
+        if (self.statement.statement_id) {
            return this._rpc({
-                   model: 'account.bank.statement.line',
+                   model: 'account.bank.statement',
                    method: 'button_confirm_bank',
-                   args: [self.bank_statement_line_id.id],
+                   args: [self.statement.statement_id],
                })
                .then(function () {
-                    return self.bank_statement_line_id.id;
+                    return self.statement.statement_id;
                });
+        }
+        return Promise.resolve(false);
    },
```
I've also notice that the template currently display the close statement button if the active_model in the context is 'account_bank_statement', dunno if it is possible that we might end up with multiple statement and an active_model of account_bank_statement or not? Anyway since this is not a full refactoring I guess we can assume that it won't happen. Otherwise we can also fix that to ensure that the button is only displayed when we have one statement.